### PR TITLE
Use a bitfield sequence to configure LED behaviour for gamepad status.

### LINKED ISSUE #61: Feature request: Allow more than four gamepads
Some games allow more than four players. It would be great if we could configure the server to allow arbitrary numbers of controllers.

Right now, this can be achieved to some extent by running multiple servers. (I had to edit a few strings to make it easy to tell controllers apart by name.) But a single server would be ideal.

### PR DESCRIPTION
**Pull request by sbuller**
- app/virtual_gamepad_hub.coffee now depends on config.json to get the number of valid gamepads.
- an additional bitField parameter is passed over the socket in the gamepadConnected message
- I did not remove the slotNumber variable and associated code from virtual_gamepad_client.js, even though it looks to be unused.
- config.json requires ledBitFieldSequence to configure the LEDs. Each array index looks up the LED configuration for the associated playerID. (both 0 indexed)
- this follows discussion and closes #61 

The bit field is ordered with the LSB on the left - corresponding with LED 1. This allows the bit field and player number to map monotonically, rather than jumping around confusingly.

### CODE REVIEW

**Comment by robsdedude** [comment]
Thank you very much for you work!

Currently I don't have much time to review your PR. I've skimmed through it, however. I liked the things I saw so far :+1: 

One thing caught my attention, however: the bitmask in `config.json` is really not a userfriendly way to configure that thing. Most people (including sys admins) probably don't "speak binary fluently", if you know what I mean. It takes a lot of explanation on our side and thought on the admins' side to get the LEDs to do what they want. However, I can't come up with a better idea right now that is userfriendly and efficient to implement. I was thinking of strings or arrays in `config.json`. But both would require some sort of parsing…

**Comment by sbuller** [comment]
What I was thinking is that the default is fine in general and it probably doesn't need to be configured by most people. Mostly config.json just seemed like a convenient place to put the array, since I wanted to refer to it from a couple places. Leaving it there strikes me as benign, and even if someone doesn't fully understand it I figure that it's simple enough to play with. Making random changes and seeing what happens is a basic tinkering skill. I'm still in favour of it as it is. If someone desperately needs to have 15 players, but I only coded for 10 - well they can just tack some 15s onto the end. Or fill in the gaps. Or repeat the array. None of that really needs much understanding of the details.

TLDR; the defaults are great, and it still feels nicer to offer the configuration than just removing the array from config.json.

File: app/virtual_gamepad_hub.coffee [hunk 1]
```diff
@@ -5,12 +5,15 @@ Virtual gamepad hub class
 
 gamepad = require './virtual_gamepad'
 log = require '../lib/log'
+config = require '../config'
+
+num_gamepads = config.ledBitFieldSequence.length
```
Comment (sbuller) [inline app/virtual_gamepad_hub.coffee]: How about I add the following comment here.
```coffeescript
# num_gamepads should be an array of 'bit fields'.
# the bit fields are numbers from 0-15, resulting in the following LED arrangements:
#  0 - ....
#  1 - *...
#  2 - .*..
#  3 - **..
#  4 - ..*.
#  5 - *.*.
#  6 - .**.
#  7 - ***.
#  8 - ...*
#  9 - *..*
# 10 - .*.*
# 11 - **.*
# 12 - ..**
# 13 - *.**
# 14 - .***
# 15 - ****
```

Comment (robsdedude) [reply app/virtual_gamepad_hub.coffee]: I'll take the comment for sure. I will then accept your PR (after I tested it, assuming it works). But I think I'll also add some info in the README any maybe a separate file, explaining `config.json`'s options would be a good thing to have anyway. So I'll probably add that as well.

Comment (sbuller) [reply app/virtual_gamepad_hub.coffee]: Sounds good. It's a shame JSON doesn't support comments.

**Comment by robsdedude** [comment]
I cannot get damn github to register this PR as merged. It is merged into develop branch now, however. Thank you for your work! For future reference: if a project has a develop branch, this is usually where to base you PR on as it might be ahead of the master and don't forget to put your changes into the changelog.
